apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/Royalty.tsx

```diff
@@ -0,0 +1,411 @@
+import { Alert, AlertTitle } from "@/components/ui/alert";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Skeleton } from "@/components/ui/skeleton";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation } from "@tanstack/react-query";
+import { CircleAlertIcon } from "lucide-react";
+import { useCallback } from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { sendAndConfirmTransaction } from "thirdweb";
+import { RoyaltyERC721 } from "thirdweb/modules";
+import { useReadContract } from "thirdweb/react";
+import { z } from "zod";
+import { addressSchema } from "../zod-schemas";
+import { ModuleCardUI, type ModuleCardUIProps } from "./module-card";
+import type { ModuleInstanceProps } from "./module-instance";
+
+function RoyaltyModule(props: ModuleInstanceProps) {
+  const { contract, ownerAccount } = props;
+
+  const defaultRoyaltyInfoQuery = useReadContract(
+    RoyaltyERC721.getDefaultRoyaltyInfo,
+    {
+      contract: contract,
+    },
+  );
+  const transferValidatorQuery = useReadContract(
+    RoyaltyERC721.getTransferValidator,
+    {
+      contract: contract,
+    },
+  );
+
+  const setRoyaltyInfoForToken = useCallback(
+    async (values: RoyaltyInfoFormValues) => {
+      if (!ownerAccount) {
+        throw new Error("Not an owner account");
+      }
+
+      const setRoyaltyForTokenTx = RoyaltyERC721.setRoyaltyInfoForToken({
+        contract: contract,
+        recipient: values.recipient,
+        bps: Number(values.bps),
+        tokenId: BigInt(values.tokenId),
+      });
+
+      await sendAndConfirmTransaction({
+        account: ownerAccount,
+        transaction: setRoyaltyForTokenTx,
+      });
+    },
+    [contract, ownerAccount],
+  );
+
+  const update = useCallback(
+    async (values: RoyaltyConfigFormValues) => {
+      if (!ownerAccount) {
+        throw new Error("Not an owner account");
+      }
+      const [defaultRoyaltyRecipient, defaultRoyaltyBps] =
+        defaultRoyaltyInfoQuery.data || [];
+
+      if (
+        values.recipient &&
+        values.bps &&
+        (values.recipient !== defaultRoyaltyRecipient ||
+          Number(values.bps) !== defaultRoyaltyBps)
+      ) {
+        const setSaleConfigTx = RoyaltyERC721.setDefaultRoyaltyInfo({
+          contract: contract,
+          royaltyRecipient: values.recipient,
+          royaltyBps: Number(values.bps),
+        });
+
+        await sendAndConfirmTransaction({
+          account: ownerAccount,
+          transaction: setSaleConfigTx,
+        });
+      }
+
+      if (
+        values.transferValidator &&
+        values.transferValidator !== transferValidatorQuery.data
+      ) {
+        const setTransferValidatorTx = RoyaltyERC721.setTransferValidator({
```

Comment:
Move updating Transfer Validator to its own function and mutation and move it to it's own section in the UI instead of combining it with the other options

Reply:
<img width="1210" alt="image" src="https://github.com/user-attachments/assets/3be1b10a-3027-4f5a-8a8b-6c3b114aad24">

There should be 2 accordions - Default Royalty info and Transfer Validator here